packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -222,7 +226,11 @@ function computePosition(
   }/* else {
     the overlay top should match the button top
   } */
-  // add the crossOffset from props
+
+  // TODO: crossAxis behaves differently from offset. A positive offset value will always shift the overlay in the "direction" it has
+  // (aka a overlay positioned "top" will be shifted upwards when a positive offset is applied and a overlay positioned "bottom" will be shifted downwards with the same offset)
+  // However, a positive crossAxis value always shifts the overlay up/to the right regardless of the cross axis direction... This means that the cross axis we apply to the submenu
+  // is incorrect if the submenu needs to flip upwards since it will shift it upwards instead of downwards... Changing this behavior is a breaking change though...
```

Comment:
Just something to note, was more pertinent when I was trying to make the submenus flip along the axis formed by their parent menu item trigger. The current behavior of `crossOffset` is different from `offset` since a positive `offset` always moves the overlay in the direction of the main axis but `crossOffset` will move the overlay "up" or to the "right" regardless of the `crossAxis` direction. I'll get rid of this comment later but just wanted to point it out for potential discussions over desired behavior

Reply:
good catch on this one, its an issue with the maxHeight calculation that needed to account for the padding, will push a fix once I figure out some unrelated Safari positioning issues.